index.js

```diff
@@ -1,34 +1,13 @@
 'use strict'
 
 const fp = require('fastify-plugin')
-const mysql = require('mysql')
 
 function fastifyMysql (fastify, options, next) {
-  const name = options.name
-  delete options.name
-
-  const pool = mysql.createPool(options.connectionString || options)
-  const db = {
-    connect: onConnect => pool.getConnection(onConnect),
-    pool: pool,
-    query: pool.query.bind(pool)
-  }
-
-  if (name) {
-    if (!fastify.mysql) {
-      fastify.decorate('mysql', {})
-    }
-
-    fastify.mysql[name] = db
+  if ((options.type || 'pool') === 'pool') {
```

Comment:
this would always be `true`.

Reply:
Yes, default use pool.

Reply:
it's true no matter what `options.type` value is

Reply:
I'll fixed 
```js
  if (options.type && options.type === 'connection') {
    fastify.register(require('./connection'), options)
  } else {
    fastify.register(require('./pool'), options)
  }
```